aten/src/ATen/native/quantized/cpu/qlinear.cpp

```diff
@@ -931,10 +931,19 @@ static at::Tensor linear_int8_with_onednn_weight(
         output_scale == 1.0f && output_zero_point == 0, "onednn qlinear: expect scale=1 and zero point=0 for fp32 output");
   }
 
-  auto input_contig = input.contiguous();
+  // If the input has more than two dimensions, we will reshape it to a 2-dimensional form
+  // for calculation and subsequently reshape the output back.
+  auto input_contig =
+      dim == 2 ? input.contiguous() : input.reshape({-1, input.size(dim - 1)});
```

Comment:
I don't think `reshape` can guarantee the contiguity of the result. Suggest to explicitly make the result contiguous. A separate question: can we support transposed input?

Reply:
> I don't think reshape can guarantee the contiguity of the result. Suggest to explicitly make the result contiguous.

Thanks for the comments and changed.

Reply:
> A separate question: can we support transposed input?

I think basically it should depend on whether oneDNN can support the calculation with transposed input, we may check with oneDNN team for that. And for now, in the inductor lowering, the BF16/INT8 Linear assumes the input and output buffer are contiguous as in: https://github.com/pytorch/pytorch/blob/8241fe6edb11278a236e61f4a87eae6be9f032d9/torch/_inductor/ir.py#L5190 and https://github.com/pytorch/pytorch/blob/8241fe6edb11278a236e61f4a87eae6be9f032d9/torch/_inductor/ir.py#L4825-L4828

So, it may be ok if the backend only support contiguous input and output.